packages/engine/Source/DataSources/FeatureLayer.js

```diff
@@ -0,0 +1,213 @@
+import Rectangle from "../Core/Rectangle.js";
+import RuntimeError from "../Core/RuntimeError.js";
+import defined from "../Core/defined.js";
+import PolygonGeometry from "../Core/PolygonGeometry.js";
+import Feature from "./Feature.js";
+import AssociativeArray from "../Core/AssociativeArray.js";
+import CustomDataSource from "./CustomDataSource.js";
+import parseGeoJson from "./parseGeoJson.js";
+import Ellipsoid from "../Core/Ellipsoid.js";
+import Cartesian3 from "../Core/Cartesian3.js";
+
+/**
+ * @import FeatureProvider from "./FeatureProvider";
+ * @import FrameState from "../Scene/FrameState";
+ * @import OGCFeaturesProvider from "./OgcFeatureProvider.js";
+ */
+
+/**
+ * A class for things
+ * @constructor
+ * @param {FeatureProvider | OGCFeaturesProvider} provider
```

Comment:
If `OGCFeaturesProvider` implements `FeatureProvieder`, I don't think we'd need to specify the type explicitly, no?

Reply:
Technically no. However as previously discussed at length TS will not recognize inheritance that's defined purely through JSDoc so it doesn't understand this without both. I have been adding JSDoc comments for types to help me while I develop but may need to do a pass on all of them before this is merged to clean them up to be "correct"